### DISCUSSION TITLE
</head> and </body> tags are missing in HTTP response. I wonder why.

### DISCUSSION THREAD

**Discussion by hyngng** [cotes2020/jekyll-theme-chirpy#1789] [Q&A] [answered]
### Checklist

- [X] I have read and accepted the [contributing guidelines](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/docs/CONTRIBUTING.md).

### How did you create the site?

Built from `jekyll-theme-chirpy`

### Description

<br>

![image](https://github.com/cotes2020/jekyll-theme-chirpy/assets/96360829/32aa2ec1-864d-4ea8-9166-ebd6c926d791)

- I noticed that my blog is missing the  closing `</head>` and `</body>` tags. I checked the [official page](https://chirpy.cotes.page/) and found the same issue. I thought it might be a bug generated in a recent update, so I checked older versions of the Chirpy theme, but the issue is still there. Is this intentional?

- Although the rendered code shows `</head>` and `</body>`, they are missing in the HTTP response.

<br>

![image](https://github.com/cotes2020/jekyll-theme-chirpy/assets/96360829/9c24860c-f256-4714-8ed2-f5f6007df9c5)
- I checked other themes such as [Minimal-Mistakes](https://mmistakes.github.io/minimal-mistakes/), and they don't have this issue, so I'm wondering why this is happening.

### Operations you have already tried

- uh, no effort yet...?

**Comment by lamininA1** [comment]
I'm not a expert but when I did some googling, you can skip some closing tags like `</head>` or `</body>` since they use HTML5 markup. It would be much better if we have closing tags though. :)

**Comment by kungfux** [comment]
That's true. Browsers are smart enough to handle such cases but that does not mean we should violate the rules.

By the way, the issue reproduces for production build only. In development mode closing tags are present as expected.

**Comment by cotes2020** [comment, marked as answer]
I bet you guys didn't know this site existed: <https://html.spec.whatwg.org/multipage/#toc-semantics>. Check out the _Tag omission_ section in the `head` and `body` tags if you're interested.

**Reply by kungfux** [reply]
Easy to learn, a lifetime to master.

However, they seem to like to overcomplicate things or I really miss something 😅

**Reply by lamininA1** [reply]
Bro that's hurt! I just saw this post. https://www.w3schools.com/html/html5_syntax.asp
Again, I'm not a expert like you guys. If you're right, it is right. 👍

**Reply by hyngng** [reply]
Thanks for the answer! My curiosity has been resolved 🙂

**Reply by sdepouw** [reply]
Glad I found this post! Just spun up a Chirpy-themed Jekyll site today, and noted their omission.
